modules/standard/IO.chpl

```diff
@@ -3088,101 +3464,140 @@ record defaultDeserializer {
       if !_first then reader._readLiteral(" ");
       else _first = false;
 
       reader.read(element);
     }
 
-    @chpldoc.nodoc
+    /*
+      End deserialization of the current array.
+    */
     proc endArray() throws {
     }
   }
 
-  // Array helpers
-  @chpldoc.nodoc
-  proc startArray(reader: fileReader) throws {
-    return new ArrayDeserializer(reader);
+  /*
+    Start deserializing a map by reading a ``{``.
+
+    :arg reader: The ``fileReader`` to use when deserializing.
+
+    :returns: A new :type:`MapDeserializer`
+  */
+  proc startMap(reader: fileReader) throws {
+    reader._readLiteral("{");
+    return new MapDeserializer(reader);
   }
 
+  /*
+    Returned by ``startMap`` to provide the API for deserializing maps.
+
+    See ``defaultSerializer.MapSerializer`` for details of the default
+    format for map.
+  */
   record MapDeserializer {
+    @chpldoc.nodoc
     var reader;
+    @chpldoc.nodoc
     var _first : bool = true;
 
-    @chpldoc.nodoc
+    /*
+      Deserialize and return a key of type ``keyType``.
+    */
     proc ref readKey(type keyType) : keyType throws {
       if !_first then reader._readLiteral(", ");
       else _first = false;
 
       return reader.read(keyType);
     }
 
-    @chpldoc.nodoc
+    /*
+      Deserialize ``key`` in-place as a key of the map.
+    */
     proc ref readKey(ref key) throws {
       if !_first then reader._readLiteral(", ");
       else _first = false;
 
       reader.read(key);
     }
 
-    @chpldoc.nodoc
+    /*
+      Deserialize and return a value of type ``valType``.
+    */
     proc readValue(type valType) : valType throws {
       reader._readLiteral(": ");
 
       return reader.read(valType);
     }
 
-    @chpldoc.nodoc
+    /*
+      Deserialize ``value`` in-place as a value of the map.
+    */
     proc readValue(ref value) throws {
       reader._readLiteral(": ");
 
       reader.read(value);
     }
 
-    @chpldoc.nodoc
+    /*
+      End deserialization of the current map by reading a ``}``.
+    */
     proc endMap() throws {
       reader._readLiteral("}");
     }
 
-    // Note: behavior undefined if called between readKey/readValue
-    @chpldoc.nodoc
+    /*
+      :returns: Returns ``true`` if there are more elements to read.
+
+      .. warning::
+
+        Behavior of 'hasMore' is undefined when called between ``readKey`` and
+        ``readValue``.
+    */
     proc hasMore() : bool throws {
       reader.mark();
       defer reader.revert();
       return !reader.matchLiteral("}");
     }
   }
-
-  // Map helpers
-  @chpldoc.nodoc
-  proc startMap(reader: fileReader) throws {
-    reader._readLiteral("{");
-    return new MapDeserializer(reader);
-  }
 }
 
 @deprecated(notes="'DefaultDeserializer' is deprecated; please use 'defaultDeserializer' instead")
 type DefaultDeserializer = defaultDeserializer;
 
 /*
-  An unstructured binary Serializer to be used with ``fileWriter``.
+  A basic binary Serializer.
```

Comment:
I feel like we shouldn't use the word "basic" (I read it as kind of demeaning).

Reply:
I can see that, but still want to use some kind of word to emphasize that this isn't a particularly robust format. Would "simple" work?

Reply:
Yeah I think simple might be a better word.